EverCigaretteUse Inference

In [3]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

In [4]:
df = pd.read_csv("YRBS_2007.csv", encoding="latin1")
pd.set_option('display.max_columns', None)
df.head()

,RaceEth,HowOldAreYou,WhatIsYourSex,InWhatGradeAreYou,AreYouHispanicOrLatino,WhatIsYourRace,HowTallAreYouWithoutShoesInMeters,HowMuchDoYouWeighWithoutShoesInKG,BicyleHelmetUse,SeatBeltUse,RidingWithADrinkingDriver,DrinkingAndDriving,WeaponCarrying,GunCarryingPast12Mos,WeaponCarryingAtSchool,SafetyConcernsAtSchool,WereThreatenedOrInjuredWithAWeaponOnSchoolProperty,StolenOrDamagedYourProperty,PhysicalFighting,InjuredIFight,PhysicalFightingAtSchool,BoyfriendGirlfriendPhysicallyHurt,ForcedSexualIntercourse,SadOrHopeless,ConsideredSuicide,MadeASuicidePlan,AttemptedSuicide,InjuriousSuicide,EverCigaretteUse,InitiationSmokingWholeCigarette,CurrentCigaretteUse,SmokedMoreThan10Cigarettes,HowObtainedCigarettes,SmokeOnSchoolProperty,EverSmokedDailyFor30Days,EverSmokingCessation,CurrentSmokelessTobaccoUse,CurrentSmokelessTobaccoOnSchoolProperty,CurrentCigarUse,EverAlcoholUse,InitiationOfAlcoholUse,CurrentAlcoholUse,CurrentBingeDrinking5OrMore,SourceOfAlcohol,DrinkAlcoholOnSchoolProperty,EverMarijuaUse,InitiationOfMarijuaUse,CurrentMarijuaUse,MarijuaOnSchoolProperty,EverCocaineUse,CocaineUsePast30Days,EverInhalantUse,EverHeroinUse,EverMethamphetamineUse,EverEcstasyUse,EverSteriodUse,IllegalInjectedDrugUse,IllegalDrugsAtSchool,EverSexualIntercourse,FirstSexualIntercourse,MultipleSexPartners,CurrentSexualActivity,AlcoholDrugsAndSex,CondomUse,BirthControlPillUse,PerceptionOfWeight,WeightLoss,ExerciseToLoseWeight,ConsumeFewerCaloriesToLoseWeight,FastingToLoseWeight,DietPillsToLoseWeight,LaxativesToLoseWeight,FruitJuiceDrinking,FruitEating,GreenSaladEating,PotatoEating,CarrotEating,OtherVegetableEating,NoSodaDrinking,NoMilkDrinking,PhysicalActivity5OrMoreDays,TelevisionWatching,ComputerUse,PEAttendance,SportsTeamParticipation,TaughtAboutHIV,Asthma,StillHaveAsthma,UsedMotorcycleHelmet,EverUsedLSD,AerobicExercise,NoerobicExercise,MinutesInPEPlayingSports,InjuredWhileExercising,HIVTesting,SunscreenUse,SunProtection,Sleep,HealthInGeneral,BMIPCT,weight,stratum,psu
0,7.0,4.0,2.0,2.0,1.0,C,NaN,NaN,2.0,1.0,5.0,5.0,4.0,3.0,1.0,5.0,6.0,3.0,8.0,1.0,1.0,1.0,1.0,1.0,2.0,2.0,1.0,1.0,1.0,2.0,7.0,7.0,2.0,3.0,1.0,3.0,5.0,4.0,4.0,NaN,2.0,NaN,4.0,5.0,3.0,NaN,2.0,NaN,NaN,5.0,2.0,4.0,6.0,4.0,4.0,3.0,1.0,2.0,1.0,6.0,5.0,4.0,2.0,3.0,2.0,3.0,3.0,1.0,1.0,2.0,2.0,2.0,5.0,5.0,3.0,2.0,1.0,1.0,7.0,5.0,5.0,4.0,2.0,1.0,1.0,1.0,2.0,3.0,2.0,2.0,6.0,8.0,1.0,3.0,2.0,1.0,1.0,5.0,3.0,NaN,1.5104,101,11030
1,5.0,7.0,2.0,2.0,2.0,E,1.70,68.04,4.0,4.0,5.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,NaN,NaN,NaN,2.0,2.0,2.0,2.0,2.0,3.0,3.0,1.0,4.0,NaN,NaN,NaN,NaN,2.0,2.0,7.0,7.0,4.0,4.0,7.0,NaN,NaN,NaN,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,2.0,1.0,7.0,7.0,2.0,2.0,2.0,4.0,3.0,1.0,1.0,2.0,2.0,2.0,2.0,4.0,4.0,4.0,4.0,4.0,4.0,4.0,4.0,2.0,6.0,3.0,2.0,2.0,1.0,3.0,4.0,1.0,1.0,2.0,2.0,1.0,2.0,3.0,1.0,5.0,4.0,3.0,66.531824,1.8559,101,11030
2,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,5.0,3.0,3.0,3.0,3.0,3.0,3.0,2.0,4.0,5.0,4.0,3.0,4.0,2.0,2.0,1.0,2.0,2.0,3.0,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,4.0,4.0,4.0,4.0,4.0,NaN,2.0,2.0,NaN,NaN,6.0,NaN,2.0,2.0,1.0,5.0,5.0,4.0,2.0,4.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2.0,3.0,2.0,2.0,1.0,1.0,2.0,4.0,4.0,2.0,2.0,4.0,2.0,2.0,1.0,3.0,4.0,3.0,3.0,2.0,2.0,2.0,2.0,NaN,3.0,2.0,4.0,4.0,2.0,3.0,2.0,1.0,1.0,1.0,NaN,1.8559,101,11030
3,7.0,1.0,1.0,1.0,1.0,A,1.63,79.38,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,NaN,1.0,1.0,1.0,1.0,NaN,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,98.174319,1.3264,101,11030
4,7.0,1.0,1.0,5.0,1.0,B,NaN,NaN,6.0,5.0,5.0,NaN,5.0,5.0,5.0,1.0,1.0,1.0,NaN,1.0,1.0,1.0,1.0,1.0,NaN,NaN,NaN,1.0,NaN,2.0,1.0,NaN,NaN,NaN,NaN,3.0,7.0,NaN,NaN,NaN,1.0,NaN,1.0,NaN,7.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,1.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,7.0,7.0,NaN,N

In [9]:
# 建立 binary（1=有抽過, 2=沒有）
df['cig_binary'] = df['EverCigaretteUse'].apply(lambda x: 1 if x == 1 else 0)

p_hat = df['cig_binary'].mean()
n = df['cig_binary'].count()

print("Sample proportion:", p_hat)
print("Sample size:", n)

Sample proportion: 0.5102200697955986
Sample size: 14041


In [11]:
p0 = 0.50
print("Benchmark proportion (p0) =", p0)

Benchmark proportion (p0) = 0.5


In [13]:
import numpy as np

se = np.sqrt(p_hat * (1 - p_hat) / n)

lower = p_hat - 1.96 * se
upper = p_hat + 1.96 * se

print("95% CI:", (lower, upper))

95% CI: (0.501951387355792, 0.5184887522354052)


In [15]:
from scipy import stats

z = (p_hat - p0) / se
p_value = 2 * (1 - stats.norm.cdf(abs(z)))

print("Z statistic:", z)
print("p-value:", p_value)

Z statistic: 2.4225548562537194
p-value: 0.01541179870653342


The sample proportion of students who have ever used cigarettes is 0.5102200697955986.
The 95% confidence interval is  (0.501951387355792, 0.5184887522354052).
The p-value is 0.01541179870653342.
Since p-value < 0.05, we reject the null hypothesis.（or ≥ 0.05 → fail to reject）
This suggests that the true proportion is different from  0.50.

In [18]:
mean = df['BMIPCT'].mean()
std = df['BMIPCT'].std()
n = df['BMIPCT'].count()

print("Mean:", mean)
print("Std:", std)
print("n:", n)

Mean: 64.82068317411306
Std: 27.516756256180642
n: 13062


In [20]:
mu0 = 65
print("Benchmark mean μ₀:", mu0)

Benchmark mean μ₀: 65


In [22]:
se = std / np.sqrt(n)

lower = mean - 1.96 * se
upper = mean + 1.96 * se

print("95% CI:", (lower, upper))

95% CI: (64.3487848135942, 65.29258153463192)


In [24]:
from scipy import stats

t_stat, p_value = stats.ttest_1samp(df['BMIPCT'].dropna(), 65)

print("t-stat:", t_stat)
print("p-value:", p_value)

t-stat: -0.7447810972515234
p-value: 0.4564174644584005


The sample mean BMI percentile is 64.82068317411306,standard deviation is 27.516756256180642 and sanple size is 13062.
The 95% confidence interval is (64.3487848135942, 65.29258153463192).
The p-value is 0.4564174644584005.
Since p-value > 0.05, we don't reject the null hypothesis.
This suggests that the mean BMI percentile is not different from  65.

Interpretation EverCigaretteUse

For the proportion analysis, we estimated the population proportion of students who have ever smoked.
We also tested whether this population proportion is different from the benchmark value of 0.50.

Because the p-value is greater than or equal to 0.05, we fail to reject the null hypothesis.
This means that there is not enough evidence to conclude that the population proportion of students who have ever smoked is different from 0.50.

This result is consistent with the EDA, which showed that the proportion of students who had ever smoked appeared to be relatively high.

The results should be interpreted cautiously because missing values, possible outliers, and the non-symmetric distribution of the data may affect the analysis.
In addition, statistical significance does not necessarily imply practical importance.

Interpretation BMIPCT

For the mean analysis, we estimated the population mean BMI percentile.
We also tested whether the population mean BMI percentile is different from the benchmark value of 65.

For BMIPCT, the sample size was  13062.
The sample mean was 64.82068317411306 and the sample standard deviation was 27.516756256180642.
The 95% confidence interval for the population mean was (64.3487848135942, 65.29258153463192).
The p-value from the one-sample t-test was 0.4564174644584005.